CMakeLists.txt

```diff
@@ -93,13 +96,26 @@ set(QUDA_INTERFACE_TIFR OFF CACHE BOOL "build tifr interface")
 #QKXTM: DMH
 set(QUDA_INTERFACE_QKXTM ON CACHE BOOL "build QKXTM interface")
 set(QUDA_QKXTM_ARPACK OFF CACHE BOOL "enable QKXTM ARPACK")
+
 if(QUDA_INTERFACE_QKXTM)
-  # Locations for GSL HDF5 LIME OPENBLAS
+  # Locations for GSL HDF5 LIME
   set(QUDA_QKXTM_GSLHOME "" CACHE PATH "path to GSL")
   set(QUDA_QKXTM_LIMEHOME "" CACHE PATH "path to LIME")
   set(QUDA_QKXTM_HDF5HOME "" CACHE PATH "path to HDF5")
-  set(QUDA_QKXTM_OPENBLAS_BUILD "" CACHE PATH "path to OPENBLAS_BUILD")
-  set(QUDA_QKXTM_OPENBLAS_SRC "" CACHE PATH "path to OPENBLAS_SRC")
+
+#-C.K. Locations for OPENBLAS or MKL, depending on GNU/OpenBlas, Intel/MKL switches
+  if(${QUDA_QKXTM_GNU-BUILD})
+    set(QUDA_QKXTM_OPENBLAS_BUILD "" CACHE PATH "path to OPENBLAS_BUILD")
+    set(QUDA_QKXTM_OPENBLAS_SRC "" CACHE PATH "path to OPENBLAS_SRC")
+  endif()
+
+#-C.K. In facts, it seems that these options do not really contribute something,
+# but may be useful for future iterations
+  if(${QUDA_QKXTM_INTEL-BUILD})
+    set(QUDA_QKXTM_MKL_LIBDIR "" CACHE PATH "path to MKL library dir")
+    set(QUDA_QKXTM_MKL_INCDIR "" CACHE PATH "path to MKL include dir")
+  endif()
+#-C.K.
```

Comment:
Yeah, I've been looking at that for a while. Everything is contained in the `blas_magma.cu` file, and you can see the slowdown caused by Magma when the code is profiled. This could be a good collaborative project for us, but it will mostly be a benefit for HMC, so not sure if it's worth it if we don't port our generation code. It's only a few routines though :)